src/cmdliner.mli

```diff
@@ -898,7 +898,9 @@ module Arg : sig
 
   val info :
     ?deprecated:string -> ?absent:string -> ?docs:string -> ?docv:string ->
-    ?doc:string -> ?env:Cmd.Env.info -> string list -> info
+    ?doc:string -> ?env:Cmd.Env.info -> 
+    ?complete:[ `Complete_custom of unit -> (string * string) list | `Complete_dir | `Complete_file ] ->
```

Comment:
We will need a proper type name for that. Also I don't think this belongs to `info` it belongs to the `'a Arg.conv` type. E.g. if I use `Arg.file` converter I shouldn't have to repeat `` `Complete_file ``. 

Currently this type is not abstract but it has been scheduled to become [since 2017](https://github.com/dbuenzli/cmdliner/blob/master/CHANGES.md#v100-2017-03-02-la-forclaz-vs). So probably it's now time to act and make `'a conv` into a proper abstract record. 

Does that still fit your approach ?